contracts/token/ERC721/ConfidentialERC721URIStorage.sol

```diff
@@ -35,31 +39,79 @@ abstract contract ConfidentialERC721URIStorage is IERC165, ConfidentialERC721 {
     function _setTokenURI(
         address to,
         uint256 tokenId,
-        gtUint64 _tokenURI
+        ctUint64[] calldata _itTokenURI,
+        bytes[] calldata _itSignature
     ) internal virtual {
-        _tokenURIs[tokenId] = MpcCore.offBoardCombined(_tokenURI, to);
+        gtUint64[] memory _tokenURI = new gtUint64[](_itTokenURI.length);
+
+        itUint64 memory it;
+
+        for (uint256 i = 0; i < _itTokenURI.length; ++i) {
+            it.ciphertext = _itTokenURI[i];
+            it.signature = _itSignature[i];
+
+            _tokenURI[i] = MpcCore.validateCiphertext(it);
```

Comment:
if validation will fail , what will happen to code?

Reply:
I guess it depends on the behaviour of validateCiphertext (this should also be documented somewhere). I assumed it will revert but I will add a test case for it.